Bottom: 85e88f5c0a21896fb7671393f42dfc25bbd7b8aa
Top:    bb89a004c44043db007a668c9f6e01fe71986845
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-04-05 22:43:26 +0100

Refresh of gtk-hid-use-clipping-region-wh

---

diff --git a/src/hid/gtk/gui-render-pixmap.c b/src/hid/gtk/gui-render-pixmap.c
index 5979cf8..657d654 100644
--- a/src/hid/gtk/gui-render-pixmap.c
+++ b/src/hid/gtk/gui-render-pixmap.c
@@ -76,6 +76,11 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
   gport->view_y0 = ghid_flip_y ? PCB->MaxHeight - cy : cy;
   gport->view_y0 -= gport->view_width  / 2;
 
+  region.X1 = gport->view_x0;
+  region.Y1 = gport->view_y0;
+  region.X2 = gport->view_x0 + gport->view_width;
+  region.Y2 = gport->view_y0 + gport->view_height;
+
   /* clear background */
   gdk_draw_rectangle (pixmap, gport->bg_gc, TRUE, 0, 0, width, height);
